### PR TITLE
feat: support kinds of messages

### PR DESCRIPTION
**Pull request by fabian4**
## 消息种类支持

消息类型 | 从属(根据接口返回) | api | 接收 | 发送 | 群聊
-- | -- | -- | -- | -- | --
文本 | text | message.text | ✅ | ✅ | ❌
图片 | image | message.toImage() | ✅ | ✅ | ❌
视频 | video | message.toFilebox() | ✅ | ❌ | ❌
音频 | audio | message.toFilebox() | ✅ | ❌ | ❌
位置 | location | message.toLocation() | ❌ | ❌ | ❌
文件 | other | message.toFilebox() | ✅ | ❌ | ❌
联系人 | other | message.toContact() | ❌ | ❌ | ❌

### CODE REVIEW

File: src/help/request.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export async function updateToken () {
   // setTimeout(updateToken, 2 * 60 * 60 * 1000)
 }
 
-export async function uploadFile (temp: boolean, file: FileBoxInterface): Promise<FileItem> {
+export async function uploadFile (temp: boolean, file: FileBoxInterface): Promise<FileItem[]> {
```
Comment (su-chang) [inline src/help/request.ts:32]: Why return an array?

File: src/help/struct.ts [hunk 2]
```diff
@@ -9,10 +9,19 @@ export interface FileItem {
 
 export interface MessageItem {
   contentEncoding: string,
-  contentText: string | FileItem,
+  contentText: string | FileItem[],
   contentType: string
 }
 
+export enum MessageRawType {
+  image = 'image',
```
Comment (su-chang) [inline src/help/struct.ts:17]: It's better to use uppercase for enum.

File: src/help/struct.ts [hunk 3]
```diff
@@ -21,7 +30,7 @@ export interface WalnutMessagePayload {
   destinationAddress: string,
   messageFileSize: number,
   messageId: string,
-  messageItem: string,
+  messageItem: MessageRawType,
```
Comment (su-chang) [inline src/help/struct.ts:33]: `messageItem` is used for message type?

Comment (fabian4) [reply src/help/struct.ts:33]: yep

**Review by su-chang — APPROVED** [review]
LGTM

**Review by lijiarui — APPROVED** [review]
LGTM

**Comment by fabian4** [comment]
> Are there possible to add some unit tests?

I am on my way to do it in the next pr.

**Comment by huan** [comment]
It's great to know that you have a plan to add unit tests.

It will be a good practice to add the unit tests with the code together in one PR because we want to make sure our code is working as expected by running the unit tests

**Comment by fabian4** [comment]
@huan ping

**Review by huan — APPROVED** [review]
LGTM

Great to see that there have two unit tests in this PR!

**Review by IronConrad — COMMENTED** [review]
thanks for all your hard working

**Review by IronConrad — APPROVED** [review]

**Pull request merged**